src/Components/Calculator.js

```diff
@@ -1,16 +1,38 @@
 import React from 'react';
 import Button from './Button';
 import Screen from './Screen';
+import calculate from '../logic/calculate';
 import './Calculator.css';
 
-class Calculator extends React.Component() {
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { sum: { total: 0, next: null, operation: null } };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick = (text) => {
+    const { sum } = this.state;
+    const sumObject = calculate(sum, text);
+    this.setState({ sum: sumObject });
+  };
+
   render() {
-    const buttons = ['AC', '+/-', '%', '+', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
+    const buttons = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
 
+    const { sum } = this.state;
+    const { total, next, operation } = sum;
     return (
       <div className="calculator">
-        <Screen className="screen" />
-        {buttons.map((button) => <Button key={button.id} text={button} />)}
+        <Screen total={total} next={next} operation={operation} className="screen" />
+        {buttons.map((button) => (
+          <Button
+            key={button.id}
```

Comment:
- [ ] The `id` attribute does not exist on the button item. This is why you have this console error asking for a "unique" key prop shown below 👇 Kindly do `key={button}` instead 👍 
![image](https://user-images.githubusercontent.com/46839548/212122251-57a748f9-04d6-4a22-8a40-c07fa4324289.png)